enhancements/proxy-passdown.md

```diff
@@ -0,0 +1,162 @@
+
+For Go-based operators, set of functions will be created in operator-lib
+(and offered upstream to controller-runtime) to read the proxy variables
+from the Operator environment and inject these variables into each of
+the container specs of various kubernetes objects. Operator authors will
+be responsible for invoking these functions. We will need functions for:
+
+   0. operatorlib.ReadProxyVarsFromEnv() // Helper function
+   1. operatorlib.AddProxyToDeployment(dep *appsv1.Deployment)
+   2. operatorlib.AddProxyToDaemonSets(daem *appsv1.DaemonSet)
+   3. operatorlib.AddProxyToReplicaSets(rep *appsv1.ReplicaSet)
+   4. operatorlib.AddProxyToPods(pod *appsv1.Pod)
+
+Simplified Code:
+
+```go
+// Read environment variables for "HTTP_PROXY, HTTPS_PROXY, and
+// NO_PROXY"
+func ReadProxyVarsFromEnv() ([]corev1.EnvVar) {
+   httpProxyEnvVar := "HTTP_PROXY"
+   httpproxy := os.Getenv("HTTP_PROXY")
+   return []corev1.EnvVar{
+       {
+           Name:  httpProxyEnvVar,
+           Value: httpproxy,
+       },
+       {
+           Name:  strings.ToLower(httpProxyEnvVar),
+           Value: httpproxy,
+       },
+    }
+}
+
+func AddProxyToDeployment(dep *appsv1.Deployment) {
+    // Create []corev1.EnvVar from Operator environment
+    proxyVars := ReadProxyVarsFromEnv()
+    // Pass proxy vars to each container
+   for _, container := range dep.Spec.Template.Spec.Containers {
+       container.Env = append(container.Env, proxyVars)
+   }
+}
+```
+
+
+Example usage with the [memcached operator](https://github.com/operator-framework/operator-sdk/blob/master/testdata/go/v3/memcached-operator/controllers/memcached_controller.go#L83)
+
+```go
+dep := r.deploymentForMemcached(memcached)
+operatorlib.AddProxyToDeployment(dep)
+```
+
+
+For Ansible-based operators, the simplest way is to lookup and pass the
+environment variables directly where the objects are created by Operator
+authors. For example, the Ansible memcached `deployment.yaml` would
+include:
+```yaml
+   env:
+      HTTPS_PROXY: {{ lookup('env', 'HTTPS_PROXY') | default('', True) }}'
+```
+
+For Helm-based operators, environment variables can be set as helm
+variables using the watches file, and passed along to operands using
+templates.
+
+```yaml
+- group: example.com
+  version: v1alpha1
+  kind: Nginx
+  chart: helm-charts/nginx
+  overrideValues:
+    httpsProxy: $HTTP_PROXY
+```
+
+### User Stories
+
+#### Ansible
+
+As an ansible-based operator author, I have documentation and an example of
+passing proxy environment variables to my operands.
+
+#### Helm
+
+As a helm-based operator author, I have documentation and an example of
+passing proxy environment variables to my operands.
+
+#### Golang
+
+As a go-based operator author, I have functions that read and
+pass proxy environment variables to various operand types as well as
+documentation for how to use them.
+
+
+## Risks
+
+1. We may not have captured all of the go utility methods, we may need to
+add more helpers later.
```

Comment:
Only see this after my previous comment. +1 on keep monitoring this and expanding the helpers.